apm-agent-core/src/main/java/co/elastic/apm/agent/bci/bytebuddy/ClassAnnotationValueOffsetMappingFactory.java

```diff
@@ -0,0 +1,58 @@
+package co.elastic.apm.agent.bci.bytebuddy;
+
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.annotation.AnnotationDescription;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.method.ParameterDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.implementation.bytecode.assign.Assigner;
+
+import javax.annotation.Nullable;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+public class ClassAnnotationValueOffsetMappingFactory implements Advice.OffsetMapping.Factory<ClassAnnotationValueOffsetMappingFactory.ClassAnnotationValueExtractor>{
+
+    @Override
+    public Class<ClassAnnotationValueExtractor> getAnnotationType() {
+        return ClassAnnotationValueExtractor.class;
+    }
+
+    @Override
+    public Advice.OffsetMapping make(ParameterDescription.InDefinedShape target, AnnotationDescription.Loadable<ClassAnnotationValueExtractor> annotation, AdviceType adviceType) {
+        return new Advice.OffsetMapping() {
+            @Override
+            public Target resolve(TypeDescription instrumentedType, MethodDescription instrumentedMethod, Assigner assigner,
+                                  Advice.ArgumentHandler argumentHandler, Sort sort) {
+                return Target.ForStackManipulation.of(getAnnotationValue(instrumentedType, annotation.loadSilent()));
+            }
+        };
+    }
+
+    @Nullable
+    private Object getAnnotationValue(TypeDescription instrumentedType, ClassAnnotationValueExtractor classAnnotationValueExtractor) {
```

Comment:
Is it exactly same code as https://github.com/elastic/apm-agent-java/blob/9498bee6d90f429fdae8c0408ed6963134d68daf/apm-agent-core/src/main/java/co/elastic/apm/agent/bci/bytebuddy/AnnotationValueOffsetMappingFactory.java#L62?

If you use `AnnotationSource` can you reuse this method?

Reply:
I deleted ClassAnnotationValueOffsetMappingFactory.